### DISCUSSION TITLE
Future imestamps drift limit - impact of disabling it?

### DISCUSSION THREAD

**Discussion by Green-Adam** [datalust/seq-tickets#2192] [Help] [answered]
I'm currently using the 30 day SEQ trial to see of SEQ will suit our needs.
I have ~15 systems without direct access to the SEQ server producing compact json log files.
These are downloaded from the systems every hour in batches from an ingestion app and logs are fed into the SEQ instance.

There are 2 issues:

1. The systems may not have correct or in sync clocks. Causing log timestamps to potentially drift from seq server time. 
2. The ~hour delay and batch download causes log events being ingested to exceed the default 57min limit of the maximumFutureTimestampDriftSeconds. My understanding is that this will swap timestamps to seq server time, which I do not want due to the log files being uploaded in batches.

Can I disable the maximumFutureTimestampDriftSeconds without significant issues performance or otherwise?

**Comment by KodrAus** [comment, marked as answer]
Hi @Green-Adam :wave: Seq is designed to work off live data, ingested as it's produced rather than on historical batches. The performance impact of historical batches is that they get split into 30 second file blocks in Seq, resulting in lots of very small files. There's also a delay before these files can be indexed, which can affect query performance. If you're not dealing with large volumes of data though then this may not be an issue in practice for you.

You can fully disable the `maximumFutureTimestampDriftSeconds` so your batches will at least keep their original timestamps by setting that value to `null` in your `Seq.json`, or by setting the `SEQ_STORAGE_MAXIMUMFUTURETIMESTAMPDRIFTSECONDS` to an empty string in Docker. We've got some docs on managing configuration [here](https://docs.datalust.co/docs/json-config#storagemaximumfuturetimestampdriftseconds).

Hope that helps!
